system/web/services/HandlerService.cfc

```diff
@@ -224,10 +234,20 @@ component extends="coldbox.system.web.services.BaseService" accessors="true"{
 	 * @return coldbox.system.web.context.EventHandlerBean
 	 */
 	function getHandlerBean( required string event ){
+		// bean already in cache?
+		if ( variables.handlerCaching && structKeyExists( handlerBeanCacheDictionary, arguments.event ) ) {
+			return handlerBeanCacheDictionary[ arguments.event ];
+		}
+
 		var handlersList 			= variables.registeredHandlers;
 		var handlersExternalList 	= variables.registeredExternalHandlers;
-		var oHandlerBean 			= new coldbox.system.web.context.EventHandlerBean( variables.handlersInvocationPath );
 		var moduleSettings 			= variables.modules;
+		var oHandlerBean 			= new coldbox.system.web.context.EventHandlerBean( variables.handlersInvocationPath );
+
+		// put bean in cache if enabled
+		if ( variables.handlerCaching ) {
```

Comment:
I will modify this slightly, because we do not want to cache ALL event handler beans because the actual event could be invalid.  This could case us to create and cache tons of unecessary beans that will never be used.

Reply:
Good point, I think there is also a thread safety issue there. Let me refactor that now.